lib/phoenix/live_dashboard/components/chart_component.ex

```diff
@@ -1,131 +1,76 @@
 defmodule Phoenix.LiveDashboard.ChartComponent do
   use Phoenix.LiveDashboard.Web, :live_component
 
-  @default_prune_threshold 1_000
-
-  @default_bucket_size 20
-
   @impl true
   def mount(socket) do
-    {:ok, socket, temporary_assigns: [data: []]}
+    {:ok, stream(socket, :data, [], dom_id: &data_dom_id/1)}
   end
 
+  defp data_dom_id(_), do: "unused"
```

Comment:
I use the new stream feature. However, because JS only expects new data—not the full list of data—in the dom we don't use the ID.